app/src/main/java/com/transloadit/android_sdk/ActivityAssembly.java

```diff
@@ -1,74 +1,123 @@
 package com.transloadit.android_sdk;
 
-import android.app.Activity;
 import android.content.SharedPreferences;
 import android.net.Uri;
 
 import com.transloadit.sdk.Assembly;
+import com.transloadit.sdk.response.AssemblyResponse;
 
 import java.io.IOException;
 import java.net.URL;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
 import java.util.Map;
 
 import io.tus.android.client.TusAndroidUpload;
 import io.tus.android.client.TusPreferencesURLStore;
 import io.tus.java.client.ProtocolException;
 import io.tus.java.client.TusClient;
-import io.tus.java.client.TusExecutor;
 import io.tus.java.client.TusUpload;
-import io.tus.java.client.TusUploader;
 
 
 public class ActivityAssembly extends Assembly {
-    Activity activity;
-    Map<String, Uri> files;
+    AssemblyProgressListener listener;
+    Map<String, Uri> fileUris;
+    List<UploadTask> uploadTasks;
+    Transloadit transloadit;
+    AssemblyStatusUpdateTask statusUpdateTask;
+    String preferenceName;
+    String url;
 
-    public ActivityAssembly(com.transloadit.sdk.Transloadit transloadit, Activity activity) {
+    public static final String DEFAULT_PREFERENCE_NAME = "tansloadit_android_sdk_urls";
+
+
+    public ActivityAssembly(Transloadit transloadit, AssemblyProgressListener listener) {
         super(transloadit);
-        this.activity = activity;
+        this.transloadit = transloadit;
+        this.listener = listener;
+        fileUris = new HashMap<String, Uri>();
+        uploadTasks = new ArrayList<UploadTask>();
+        preferenceName = DEFAULT_PREFERENCE_NAME;
+    }
+
+    @Override
+    public Transloadit getClient() {
+        return this.transloadit;
+    }
+
+    @Override
+    public int getFilesCount() {
+        return fileUris.size();
+    }
+
+    public void setPreferenceName(String name) {
+        preferenceName = name;
     }
 
     public void addFile(String name, Uri fileUri) {
-        files.put(name, fileUri);
+        fileUris.put(name, fileUri);
+    }
+
+    public void pauseUpload() {
+        for (UploadTask task : uploadTasks) {
+            task.cancel(false);
+        }
     }
 
+    public void resumeUpload() throws IOException, ProtocolException {
+        processTusFiles(url);
+    }
+
+    void onUploadFinished() {
+        listener.onUploadFinished();
+        startStatusUpdateTask();
+    }
+
+    void onFinished(AssemblyResponse response) {
+        listener.onAssemblyFinished(response);
+    }
+
+    void onUploadFailed(Exception exception) {
+        listener.onUploadFailed(exception);
+    }
+
+    void onStatusUpdateFailed(Exception exception) {
+        listener.onAssemblyStatusUpdateFailed(exception);
+    }
+
+    @Override
     protected void processTusFiles(String assemblyUrl) throws IOException, ProtocolException {
+        url = assemblyUrl;
         tusClient = new TusClient();
```

Comment:
I don't understand why it's using the tus client here directly? Shouldn't the Android SDK just use the Tus client indirectly through the Java SDK?